src/fsevent.rs

```diff
@@ -347,6 +347,7 @@ impl FsEventWatcher {
         if !path.exists() {
             return Err(Error::path_not_found().add_path(path.into()));
         }
+        let canonical_path = path.to_path_buf().canonicalize()?;
```

Comment:
Can we catch this here when it's `value: Os { code: 2, kind: NotFound, message: "No such file or directory" }`, so we don't emit anything in that case ? Would be cleaner I think.

Reply:
Are you proposing to panic on other `io::Error` variants? Unfortunately the [docs](https://doc.rust-lang.org/std/fs/fn.canonicalize.html#errors) are a bit vague about what other variants are possible, but there's also `NotADirectory` which could happen if the caller accidentally supplied a malformed path ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=036fd36ddcdcf3ce69475a893aed366c)). Granted, that example should likely result in a `path.exists()` bail, but there may be a pathological case I'm not thinking of which passes that but not `canonicalize`. 🤷 

The other route I had considered when implementing this was to mirror the behavior of `FsEventWatcher::remove_path`:

https://github.com/notify-rs/notify/blob/b3f2d77e5c66e896b7a133fabd649b143bfe8b9d/src/fsevent.rs#L334-L338

whereby the original, un-canonicalized path is used as a fallback if there's an error in `canonicalize`.

Reply:
No, I was thinking about not emitting anything in that case and just aborting this event. But the more I think about it, the more it's probably the best if we let the user know about it. Even though it is a little bit confusing and most people probably aren't interested in knowing that something happened too fast to even record the exact event.